File: packages/core/src/lib/nodes/definitions/pytorch/empty.js
/**
 * PyTorch Empty/Placeholder Node Definition
 */
import { PassthroughNodeDefinition } from '../../base';
import { BackendFramework } from '../../contracts';
export class EmptyNode extends PassthroughNodeDefinition {
    constructor() {
        super(...arguments);
        Object.defineProperty(this, "metadata", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: {
                type: 'empty',
                label: 'Empty Node',
                category: 'utility',
                color: 'var(--color-gray)',
                icon: 'Placeholder',
                description: 'Placeholder node',
                framework: BackendFramework.PyTorch
            }
        });
        Object.defineProperty(this, "configSchema", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: [
                {
                    name: 'label',
                    label: 'Custom Label',
                    type: 'text',
                    default: 'Empty Node',
                    placeholder: 'Enter custom label...',
                    description: 'Custom label for this node'
                },
                {
                    name: 'note',
                    label: 'Note',
                    type: 'text',
                    placeholder: 'Add notes here...',
                    description: 'Notes or comments about this placeholder'
                }
            ]
        });
    }
}

File: python/vision_forge/services/nodes/frameworks/tensorflow/nodes/empty.py
"""TensorFlow Empty/Placeholder Node Definition"""

from typing import Any, Dict, List, Optional

from ....core.types import ConfigField, Framework, NodeMetadata, TensorShape
from ..base import TensorFlowNodeDefinition


class EmptyNode(TensorFlowNodeDefinition):
    """
    Placeholder node for architecture planning.

    Passes input through unchanged.
    """

    @property
    def metadata(self) -> NodeMetadata:
        return NodeMetadata(
            type="empty",
            label="Empty",
            category="utility",
            color="var(--color-gray)",
            icon="Circle",
            description="Placeholder for architecture planning",
            framework=Framework.TENSORFLOW
        )

    @property
    def config_schema(self) -> List[ConfigField]:
        return [
            ConfigField(
                name="note",
                label="Note",
                type="text",
                description="Notes about this placeholder"
            )
        ]

    def compute_output_shape(
        self,
        input_shape: Optional[TensorShape],
        config: Dict[str, Any]
    ) -> Optional[TensorShape]:
        return input_shape

    def validate_incoming_connection(
        self,
        source_node_type: str,
        source_output_shape: Optional[TensorShape],
        target_config: Dict[str, Any]
    ) -> Optional[str]:
        return None
